Add release notes, update authors and updating (#5613)
microsoft/MixedRealityToolkit-Unity: Documentation/Updating.md

# Updating to the 2.0.0 Release

> Between the RC2 and 2.0.0 releases of the Microsoft Mixed Reality Toolkit, changes were made that may impact existing projects. This document describes those changes and how to update projects to the 2.0.0 release.

Reviewer: Should we tell people here how to upgrade? I.e do they delete the old MRTK directories and then import package?

Reviewer: Also adding on to the list in teams, maybe also add "Review your MRTK profiles in inspector"? Per comment below, DefaultRaycastProvider wasn't set

Reviewer 2: In our RC2.1 release notes, we had upgrade instructions, it was one of our learnings to provide upgrade instructions going forward.

Reviewer 2: Here is my suggested upgrade instructions. I took our RC2.1 instructions, followed them, and added instructions that were missing:

1. Close Unity
2. Delete all MixedRealityToolkit folders from RC1: MixedRealityToolkit, MixedRealityToolkit.Examples, MixedRealityToolkit.Providers, MixedRealityToolkit.SDK, MixedRealityToolkit.Services
3. Delete your Library folder
4. Import the new unity packages (Foundation is required, tools, extensions, and examples are optional). Import Foundation first.
5. In your scenes, delete the MixedRealityToolkit and MixedRealityPlaySpace objects
6. MixedRealityToolkit -> Add to Scene and Configure…

Author: added

- [API changes](#api-changes)
- [Assembly name changes](#assembly-name-changes)

## API changes

> Since the release of RC2, there have been a number of API changes including some that may break existing projects. The following sections describe the changes that have occurred between the RC2 and 2.0.0 releases.

Reviewer: At line 24, can you fix the broken link?
To access extension services, use [`MixedRealityServiceRegistry.TryGetService<T>`].

Author: done

- [MixedRealityToolkit](#mixedrealitytoolkit)

> - [Event System](#event-system)

Reviewer: The TOC list here also doesn't match up with the actual headers shown on the right hand side

Author: @Troy-Ferrell, not sure what you mean here, can you give an example? thanks!

Reviewer: There is that list of internal hyperlinks that describes the sections but it doesn't exactly match up with the list of sections outlined on the right: 

![image](https://user-images.githubusercontent.com/25975362/62982755-1db26580-bde2-11e9-9944-a789a884750c.png)

Author: i see what you mean now... thanks!

Author: added missing headings

{
    private void OnEnable()
    {

>         InputSystem?.Register(2.0.0meObject);

Reviewer 3: Find and replace `ga` :P

```suggestion
        InputSystem?.Register(gameObject);
```

Author: yep :)

    }

    private void OnDisable()
    {

>         InputSystem?.Unregister(2.0.0meObject);

Reviewer 3: ```suggestion
        InputSystem?.Unregister(gameObject);
```

Author: search / replace bit me.... :)

- `CloseDistance` public property deprecated and has been renamed to `ClosestDistance`
- Default value for `RaycastDirectionMode` is now `TrackedTargetForward` which raycasts in the direction of the tracked target transform forward
- `OrientationMode` enum values, `Vertical` and `Full`, have been renamed to `TrackedTarget` and `SurfaceNormal` respectively

> - `KeepOrientationVertical` public property has been added to control whether orientation of associated 2.0.0meObject remains vertical

Reviewer 3: ```suggestion
- `KeepOrientationVertical` public property has been added to control whether orientation of associated GameObject remains vertical
```
